src/connection.rs

```diff
@@ -207,4 +217,59 @@ impl Connection {
             },
         }
     }
+
+    fn send_query(&self, tree: &Tree) -> Result<(), Error> {
+        let token = self.query_token.get();
+        // Increment the token for the next request.
+        self.query_token.set(token.wrapping_add(1));
```

Comment:
I'm going to keep the wrapping_add, just for simplicity.  You can wrap the type in a Wrapping struct, but there's a lot more boilerplate.   You can play around with it here:

https://play.rust-lang.org/?gist=81e3d3ac27a87b2a9ad451f774de8cbd&version=stable&backtrace=0